marketpulse/devices/forms.py

```diff
@@ -0,0 +1,41 @@
+from django import forms
+
+from models import Device
+
+
+class DeviceForm(forms.ModelForm):
+
+    class Meta:
+        model = Device
+        fields = ['model', 'manufacturer', 'is_fxos']
+
+    def save(self, *args, **kwargs):
+        query = {
+            "model__iexact": self.cleaned_data['model'],
+            "manufacturer__iexact": self.cleaned_data['manufacturer']
+        }
+
+        qs = Device.objects.filter(**query)
+        if qs.exists():
+            return qs[0]
+
+        device = super(DeviceForm, self).save(*args, **kwargs)
+        device.is_fxos = False
+        device.save()
+        return device
+
+    def clean(self, *args, **kwargs):
+        cleaned_data = super(DeviceForm, self).clean()
+        model = cleaned_data.get('model')
+        manufacturer = cleaned_data.get('manufacturer')
+        is_fxos = cleaned_data.get('is_fxos')
+        cleaned_data['is_fxos'] = bool(is_fxos)
+
+        if not is_fxos:
+            msg = 'This field is required'
+            if not model:
+                self._errors['model'] = self.error_class([msg])
+            if not manufacturer:
+                self._errors['manufacturer'] = self.error_class([msg])
```

Comment:
Really minor but you can you use `any` to check if either of the two fields is missing. Probably this will make the clean method much shorter. Feel free to ignore.